src/test/getarg_tests.cpp

```diff
@@ -40,264 +31,277 @@ void LocalTestingSetup :: ResetArgs(const std::string& strArg)
-    BOOST_CHECK_EQUAL(m_local_args.GetArg("-foo", ""), "verbose");
-    BOOST_CHECK_EQUAL(m_local_args.GetIntArg("-bar", 0), 1);
+    ResetArgs(local_args, "--foo=verbose --bar=1");
+    BOOST_CHECK_EQUAL(local_args.GetArg("-foo", ""), "verbose");
+    BOOST_CHECK_EQUAL(local_args.GetIntArg("-bar", 0), 1);
 }
 
 BOOST_AUTO_TEST_CASE(boolargno)
 {
+    ArgsManager local_args;
+
     const auto foo = std::make_pair("-foo", ArgsManager::ALLOW_ANY);
     const auto bar = std::make_pair("-bar", ArgsManager::ALLOW_ANY);
-    SetupArgs({foo, bar});
-    ResetArgs("-nofoo");
-    BOOST_CHECK(!m_local_args.GetBoolArg("-foo", true));
-    BOOST_CHECK(!m_local_args.GetBoolArg("-foo", false));
-
-    ResetArgs("-nofoo=1");
-    BOOST_CHECK(!m_local_args.GetBoolArg("-foo", true));
-    BOOST_CHECK(!m_local_args.GetBoolArg("-foo", false));
-
-    ResetArgs("-nofoo=0");
-    BOOST_CHECK(m_local_args.GetBoolArg("-foo", true));
-    BOOST_CHECK(m_local_args.GetBoolArg("-foo", false));
-
-    ResetArgs("-foo --nofoo"); // --nofoo should win
-    BOOST_CHECK(!m_local_args.GetBoolArg("-foo", true));
-    BOOST_CHECK(!m_local_args.GetBoolArg("-foo", false));
-
-    ResetArgs("-nofoo -foo"); // foo always wins:
-    BOOST_CHECK(m_local_args.GetBoolArg("-foo", true));
-    BOOST_CHECK(m_local_args.GetBoolArg("-foo", false));
+    SetupArgs(local_args, {foo, bar});
+    ResetArgs(local_args, "-nofoo");
+    BOOST_CHECK(!local_args.GetBoolArg("-foo", true));
+    BOOST_CHECK(!local_args.GetBoolArg("-foo", false));
+
+    ResetArgs(local_args, "-nofoo=1");
+    BOOST_CHECK(!local_args.GetBoolArg("-foo", true));
+    BOOST_CHECK(!local_args.GetBoolArg("-foo", false));
+
+    ResetArgs(local_args, "-nofoo=0");
+    BOOST_CHECK(local_args.GetBoolArg("-foo", true));
+    BOOST_CHECK(local_args.GetBoolArg("-foo", false));
+
+    ResetArgs(local_args, "-foo --nofoo"); // --nofoo should win
+    BOOST_CHECK(!local_args.GetBoolArg("-foo", true));
+    BOOST_CHECK(!local_args.GetBoolArg("-foo", false));
+
+    ResetArgs(local_args, "-nofoo -foo"); // foo always wins:
+    BOOST_CHECK(local_args.GetBoolArg("-foo", true));
+    BOOST_CHECK(local_args.GetBoolArg("-foo", false));
 }
 
 BOOST_AUTO_TEST_CASE(logargs)
 {
+    ArgsManager local_args;
+
     const auto okaylog_bool = std::make_pair("-okaylog-bool", ArgsManager::ALLOW_ANY);
     const auto okaylog_negbool = std::make_pair("-okaylog-negbool", ArgsManager::ALLOW_ANY);
     const auto okaylog = std::make_pair("-okaylog", ArgsManager::ALLOW_ANY);
     const auto dontlog = std::make_pair("-dontlog", ArgsManager::ALLOW_ANY | ArgsManager::SENSITIVE);
-    SetupArgs({okaylog_bool, okaylog_negbool, okaylog, dontlog});
-    ResetArgs("-okaylog-bool -nookaylog-negbool -okaylog=public -dontlog=private");
+    SetupArgs(local_args, {okaylog_bool, okaylog_negbool, okaylog, dontlog});
+    ResetArgs(local_args, "-okaylog-bool -nookaylog-negbool -okaylog=public -dontlog=private");
+
+    LogInstance().StartLogging();
```

Comment:
CI fails sequentially. Maybe due to this?

Reply:
This line turns off the logger buffering feature: https://github.com/ryanofsky/bitcoin/blob/55059269fe7379912b315a80f3a1a17b33a3e23d/src/logging.cpp#L68

so that https://github.com/ryanofsky/bitcoin/blob/55059269fe7379912b315a80f3a1a17b33a3e23d/src/logging.cpp#L267 is `false` and the logger callbacks are called

https://github.com/ryanofsky/bitcoin/blob/55059269fe7379912b315a80f3a1a17b33a3e23d/src/logging.cpp#L278-L280

I have tried to run locally `src/test/test_bitcoin --log_level=all --run_test=getarg_tests` and it passes. However, ` src/test/test_bitcoin` does not pass for me so you are right and it can be reasonably reproduced. Also as one may expect variable `s` is empty (https://github.com/ryanofsky/bitcoin/blob/55059269fe7379912b315a80f3a1a17b33a3e23d/src/test/getarg_tests.cpp#L307).

Reply:
So the error is because of this line https://github.com/ryanofsky/bitcoin/blob/55059269fe7379912b315a80f3a1a17b33a3e23d/src/logging.cpp#L53 as `m_print_to_file` is `true` sometimes so the callback is not called.

Reply:
https://github.com/kiminuo/bitcoin/commit/1ca557e57d0bdfb424ce4bf8bd7a1dd76b58be35 might actually fix the issue but I'm not sure what you would say about the change. Anyway, I don't really know about a different solution so either that or closing #24375. 

This uncovers that working with static variables is hard :(

Reply:
My understanding is that Logger instance is shared between tests. So then `fs_tests` and `getarg_tests` interferes because the `BasicTestingSetup` fixture sets a file name, then `getarg_tests/logargs` and its `LogInstance().StartLogging();` does not turn off buffering of log messages and then the callback in `logargs` is not actually called.

Reply:
@ryanofsky Do you feel like switching to `BasicTestingSetup` is a good enough result for now or do you feel like it would be actually fruitful to remove the dependency on `BasicTestingSetup`?

I think it is reasonable to reserve the later for a future PR as it seems it requires more changes.

Reply:
> @ryanofsky Do you feel like switching to `BasicTestingSetup` is a good enough result for now or do you feel like it would be actually fruitful to remove the dependency on `BasicTestingSetup`?

I don't think there's a problem with depending on `BasicTestingSetup` here, but maybe I am coming at this from different perspective. I think global variables are almost always a problem, but test fixtures are only sometimes a problem (https://github.com/bitcoin/bitcoin/pull/22155#discussion_r673051693 was a place where I previously objected to using test fixtures unnecessarily).